ragna/assistants/_ai21labs.py

```diff
@@ -41,17 +74,23 @@ async def answer(
                 "numResults": 1,
                 "temperature": 0.0,
                 "maxTokens": max_new_tokens,
-                "messages": [
-                    {
-                        "text": prompt,
-                        "role": "user",
-                    }
-                ],
-                "system": self._make_system_content(sources),
+                "messages": self._render_prompt(prompt),
+                "system": system_prompt,
             },
         ) as stream:
             async for data in stream:
-                yield cast(str, data["outputs"][0]["text"])
+                yield data
+
+    async def answer(
+        self, messages: list[Message], *, max_new_tokens: int = 256
+    ) -> AsyncIterator[str]:
+        message = messages[-1]
+        async for data in self.generate(
+            [message],
+            system_prompt=self._make_system_content(message.sources),
```

Comment:
Given that we only use this function here and the system prompt is very specific to the RAG use case, should we just inline the function? Otherwise, we need to rename it to something like `_make_rag_system_content` to avoid confusion.

This applies to all assistant in this PR.

Reply:
My personal preference is to rename the function but keep it as a function call - I do see that if someone is calling a custom rag-like call outside of answer, they're perhaps likely to want to customize that piece as well, so who knows how it gets directly reused, but it does make things like subclassing a touch cleaner. I'll push a commit with the name change in  a moment